dialog/fileicon.go

```diff
@@ -0,0 +1,120 @@
+package dialog
+
+import (
+	"image/color"
+	"path/filepath"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/theme"
+	"fyne.io/fyne/widget"
+)
+
+const (
+	fileIconSize = 64
+	fileTextSize = 24
+)
+
+type fileIcon struct {
+	widget.BaseWidget
+	parent  *fileDialog
```

Comment:
The name `parent` suggests a line of ancestors (dir->dir->dir->file) which does not exist.
I would prefer `dialog` to refer to the dialog.